packages/backend-common/src/s3.ts

```diff
@@ -57,28 +70,52 @@ export const getFile = async (
 		);
 
 		const body = ReadableBody.parse(data.Body);
-
-		const stream = body.pipe(createWriteStream(destinationPath));
-
-		await new Promise<void>((resolve, reject) => {
-			stream
-				.on('finish', () => {
-					console.log(` pipe done `);
-					resolve();
-				})
-				.on('error', (error) => {
-					console.log(`Failed to writing the S3 object ${key} into file`);
-					reject(error);
-				});
-		});
-		console.log('successfully retrieved file from S3 into ', destinationPath);
+		await downloadS3Data(body, destinationPath, key);
 		return destinationPath;
 	} catch (e) {
 		console.error(e);
 		throw e;
 	}
 };
 
+export const getObjectWithPresignedUrl = async (
+	presignedUrl: string,
+	key: string,
+	workingDirectory: string,
+) => {
+	const destinationPath = `${workingDirectory}/${path.basename(key)}`;
+	const response = await axios.get<ReadStream>(presignedUrl, {
+		responseType: 'stream',
+	});
+	const body = ReadableBody.parse(response.data);
+	await downloadS3Data(body, destinationPath, key);
+	return destinationPath;
+};
+
+const downloadS3Data = async (
+	data: Readable,
+	destinationPath: string,
+	key: string,
+) => {
+	const body = ReadableBody.parse(data);
```

Comment:
It seems that we have already parsed the sream into `ReadableBody` on line 90. Probably this is redundant here?